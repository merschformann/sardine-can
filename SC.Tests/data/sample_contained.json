{
    "name": "jsonTest",
    "containers": [
        {
            "id": 0,
            "length": 600,
            "width": 400,
            "height": 300
        },
        {
            "id": 1,
            "length": 600,
            "width": 400,
            "height": 300
        },
        {
            "id": 2,
            "length": 500,
            "width": 400,
            "height": 300
        },
        {
            "id": 3,
            "length": 300,
            "width": 400,
            "height": 250
        }
    ],
    "pieces": [
        {
            "id": 0,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 1
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 304,
                    "width": 336,
                    "height": 140
                }
            ]
        },
        {
            "id": 1,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 122,
                    "width": 245,
                    "height": 367
                }
            ]
        },
        {
            "id": 2,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 392,
                    "width": 145,
                    "height": 152
                }
            ]
        },
        {
            "id": 3,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 271,
                    "width": 214,
                    "height": 393
                }
            ]
        },
        {
            "id": 4,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 351,
                    "width": 398,
                    "height": 287
                }
            ]
        },
        {
            "id": 5,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 351,
                    "width": 245,
                    "height": 287
                }
            ]
        },
        {
            "id": 6,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 1
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 294,
                    "width": 234,
                    "height": 376
                }
            ]
        },
        {
            "id": 7,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 0
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 241,
                    "width": 78,
                    "height": 115
                }
            ]
        },
        {
            "id": 8,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 1
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 154,
                    "width": 395,
                    "height": 274
                }
            ]
        },
        {
            "id": 9,
            "flags": [
                {
                    "flagId": 0,
                    "flagValue": 1
                }
            ],
            "cubes": [
                {
                    "x": 0,
                    "y": 0,
                    "z": 0,
                    "length": 60,
                    "width": 183,
                    "height": 170
                }
            ]
        }
    ],
    "rules": {
        "flagRules": [
            {
                "flagId": 0,
                "ruleType": 0,
                "parameter": 0
            }
        ]
    }
}