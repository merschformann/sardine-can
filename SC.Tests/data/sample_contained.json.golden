{
  "containers": [
    {
      "id": 0,
      "length": 600,
      "width": 400,
      "height": 300,
      "assignments": [
        {
          "piece": 6,
          "position": {
            "x": 0,
            "y": 0,
            "z": 0,
            "a": 90,
            "b": 0,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 294,
              "width": 376,
              "height": 234
            }
          ]
        },
        {
          "piece": 9,
          "position": {
            "x": 0,
            "y": 0,
            "z": 234,
            "a": 0,
            "b": 90,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 170,
              "width": 183,
              "height": 60
            }
          ]
        },
        {
          "piece": 0,
          "position": {
            "x": 294,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 0,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 304,
              "width": 336,
              "height": 140
            }
          ]
        },
        {
          "piece": 8,
          "position": {
            "x": 294,
            "y": 0,
            "z": 140,
            "a": 0,
            "b": 90,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 274,
              "width": 395,
              "height": 154
            }
          ]
        }
      ]
    },
    {
      "id": 1,
      "length": 600,
      "width": 400,
      "height": 300,
      "assignments": [
        {
          "piece": 4,
          "position": {
            "x": 0,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 0,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 351,
              "width": 398,
              "height": 287
            }
          ]
        },
        {
          "piece": 3,
          "position": {
            "x": 351,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 90,
            "c": 90
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 214,
              "width": 393,
              "height": 271
            }
          ]
        }
      ]
    },
    {
      "id": 2,
      "length": 500,
      "width": 400,
      "height": 300,
      "assignments": [
        {
          "piece": 5,
          "position": {
            "x": 0,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 0,
            "c": 0
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 351,
              "width": 245,
              "height": 287
            }
          ]
        },
        {
          "piece": 1,
          "position": {
            "x": 0,
            "y": 245,
            "z": 0,
            "a": 90,
            "b": 0,
            "c": 90
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 367,
              "width": 122,
              "height": 245
            }
          ]
        }
      ]
    },
    {
      "id": 3,
      "length": 300,
      "width": 400,
      "height": 250,
      "assignments": [
        {
          "piece": 2,
          "position": {
            "x": 0,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 0,
            "c": 90
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 145,
              "width": 392,
              "height": 152
            }
          ]
        },
        {
          "piece": 7,
          "position": {
            "x": 145,
            "y": 0,
            "z": 0,
            "a": 0,
            "b": 0,
            "c": 90
          },
          "cubes": [
            {
              "x": 0,
              "y": 0,
              "z": 0,
              "length": 78,
              "width": 241,
              "height": 115
            }
          ]
        }
      ]
    }
  ],
  "offload": []
}
